File: codechef/contest/AdaMatrix.py
if __name__ == '__main__':
    try:
        T = int(input())
        for _ in range(T):
            matrix = []
            n = int(input())
            for _ in range(n):
                submatrix = list(map(int, input().split()))
                matrix.append(submatrix)
            number_of_ways = 0
            for i in range(n-1, 0, -1):
                print(i, matrix[i][i], matrix[i])
                if matrix[i][i] != matrix[i][i-1]+1:
                    number_of_ways += 1
                    next = i + 1
                    for j in range(next):
                        for k in range(j, next):
                            print("Value of j and k: ", j, k)
                            matrix[j][k], matrix[k][j] = matrix[k][j], matrix[j][k]
                            print(matrix)
                else:
                    continue
                print(matrix)

            print(number_of_ways)
    except Exception:
        pass

File: codechef/contest/AdaMatrix.js
function solution(matrix) {
    let count = 0
    for (let i = matrix.length-1; i > 0; i--) {
        if (matrix[i][i] != matrix[i][i-1]+1){
            count++
            next = i + 1
            for (let j = 0; j < next; j++){
                for (let k = j; k < next; k++){
                    matrix[j][k], matrix[k][j] = matrix[k][j], matrix[j][k]
                }
            }
        }
        else{
            continue
        }
    }
    return count
}

let matrix = [[1,2,9,13], [5,6,10,14], [3,7,11,15], [4,8,12,16]]
console.log(solution(matrix));